### PR TITLE
Replace PropertyGraph in cugraph-PyG with FeatureStore

### LINKED ISSUE #3079: [FEA] cuGraph-PyG Should Pass Canonical Edge Type to New Uniform Neighbor Sample Function
Currently, cuGraph-PyG only passes the cuGraph edge type, but this adds unnecessary overhead.  Instead, the canonical edge type should be passed as an integer.  A lookup table may need to be added in `CuGraphStore` to handle this.

Partially resolved by #3101

### LINKED ISSUE #3151: [IMP] Migrate all PropertyGraph code in cugraph-PyG to FeatureStore
cuGraph-PyG is no longer relying on PropertyGraph for functionality.  All reliance on PropertyGraph should be replaced with calls to FeatureStore.

Due: 1/18/2023 6:00 PM EST

### PR DESCRIPTION
**Pull request by alexbarghi-nv**
Replaces PropertyGraph in cugraph-PyG with the new FeatureStore, which is the preferred solution for storing tensor features.
Closes #3151 
Closes #3079

### CODE REVIEW

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py [hunk 1]
```diff
@@ -261,140 +245,143 @@ def __init__(self, G, backend: str = "torch", renumber_graph: bool = None):
...
+            {
+                "src": na_src,
+                "dst": na_dst,
+                # FIXME use the edge type property
+                # "w": np.zeros(len(na_src)),
+                "w": na_etp,
+                "eid": np.arange(len(na_src)),
+                "etp": na_etp,
+            }
+        )
```
Comment (VibhuJawa) [inline python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py:354]: Does this mean that you always have to create a pandas dataframe for using PyG. I think we should allow a path to provide cudf and dask_cudf dataframes directly as that will probably be much more scalable.

File: python/cugraph-pyg/cugraph_pyg/tests/mg/test_mg_cugraph_sampler.py [hunk 2]
```diff
@@ -148,81 +115,100 @@ def test_neighbor_sample(basic_property_graph_1):
...
-        num_unique_sampled_edges = len(
-            cudf.DataFrame({"src": srcs, "dst": dsts}).drop_duplicates()
+    for node_type, node_ids in noi_groups.items():
+        actual_vertex_ids = cupy.arange(N[node_type])
+
+        assert list(node_ids) == list(actual_vertex_ids)
+
+    print("row:", row_dict)
+    print("col:", col_dict)
+    print("G:", G)
```
Comment (VibhuJawa) [inline python/cugraph-pyg/cugraph_pyg/tests/mg/test_mg_cugraph_sampler.py:190]:
```suggestion

```

File: python/cugraph-pyg/cugraph_pyg/sampler/cugraph_sampler.py [hunk 3]
```diff
@@ -140,12 +142,10 @@ def __neighbor_sample(
...
 
-        # FIXME eventually get uniform neighbor sample to accept longs
         if backend == "torch" and not index.is_cuda:
             index = index.cuda()
 
-        # FIXME resolve the directed/undirected issue
-        G = self.__graph_store._subgraph([et[1] for et in edge_types])
+        G = self.__graph_store._subgraph(edge_types)
 
         index = cudf.from_dlpack(index.__dlpack__())
```
Comment (VibhuJawa) [inline python/cugraph-pyg/cugraph_pyg/sampler/cugraph_sampler.py:150]: I would suggest using the to_dlpack object here or using `cp.asarray(a)` . 

https://pytorch.org/docs/stable/dlpack.html#torch.utils.dlpack.to_dlpack

```suggestion
        index = cudf.from_dlpack(index.__dlpack__())
```

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py [hunk 4]
```diff
@@ -618,29 +597,26 @@ def _get_vertex_groups_from_sample(self, nodes_of_interest: cudf.Series) -> dict
...
-                    self.searchsorted(
-                        self.from_dlpack(
-                            self.__vertex_type_offsets["stop"].to_dlpack()
-                        ),
-                        nodes_of_interest,
-                    ).__dlpack__()
-                )
+            noi_type_indices = self.searchsorted(
+                self.from_dlpack(self.__vertex_type_offsets["stop"].__dlpack__()),
+                nodes_of_interest,
```
Comment (VibhuJawa) [inline python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py:606]: Same suggestion here.

File: python/cugraph-pyg/cugraph_pyg/tests/mg/test_mg_cugraph_store.py [hunk 5]
```diff
@@ -220,283 +158,292 @@ def test_edge_attr():
...
+
+
+def test_renumber_vertices_multi_edge_multi_vertex(
+    multi_edge_multi_vertex_graph_1, dask_client
+):
+    F, G, N = multi_edge_multi_vertex_graph_1
+    cugraph_store = CuGraphStore(F, G, N, backend="cupy", multi_gpu=True)
+
+    nodes_of_interest = cudf.from_dlpack(
+        cupy.random.randint(0, sum(N.values()), 3).__dlpack__()
```
Comment (VibhuJawa) [inline python/cugraph-pyg/cugraph_pyg/tests/mg/test_mg_cugraph_store.py:236]: Same suggestion here.

**Review by VibhuJawa — CHANGES_REQUESTED** [review]

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py (hunk 1 above)
Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py:354]: I have a separate issue to allow passing in a cudf or dask_cudf dataframe.  That won't be part of this PR.

Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py:354]: See #3176

File: python/cugraph-pyg/cugraph_pyg/sampler/cugraph_sampler.py (hunk 3 above)
Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/sampler/cugraph_sampler.py:150]: This doesn't work for cudf

Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/sampler/cugraph_sampler.py]: resolved, all dlpack calls removed

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py (hunk 4 above)
Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py]: resolved, all dlpack calls removed

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py [hunk 6]
```diff
@@ -19,33 +19,37 @@
...
-)
-StructuralGraphType = Union[CuGraphGraph, CGSGraph]
+
+
+def torch_as_array(a):
+    if isinstance(a, cupy.ndarray):
+        t = torch.from_dlpack(a)
+    elif isinstance(a, cudf.Series):
+        t = torch.from_dlpack(a.to_dlpack())
+    return t
```
Comment (VibhuJawa) [inline python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py:52]: Motivation of this fix . Should have been clearer, this gives you managed memory for free with Pytorch as dlpack  fails .   

```python
import rmm      
import torch
import cudf
rmm.reinitialize(managed_memory= True)
torch.cuda.memory.change_current_allocator(rmm.rmm_torch_allocator)
```
Below works:
```python
ar = cudf.Series([1]).repeat(1_000_000_000).values
t1 = torch.as_tensor(ar, device='cuda')
t1
tensor([1, 1, 1,  ..., 1, 1, 1], device='cuda:0')
```
Below Fails: 
```python
ar = cudf.Series([1]).repeat(1_000_000_000).values
t2 = torch.from_dlpack(ar.toDlpack())
```
```python
RuntimeError: Unsupported device_type: 13
```

**Review by VibhuJawa — CHANGES_REQUESTED** [review]
Minor review on using `cuda-array-interface`

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py (hunk 6 above)
Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py:52]: I thought I changed this

Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py]: @VibhuJawa the push failed, please review again.

**Review by VibhuJawa — APPROVED** [review]
LGTM

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py (hunk 6 above)
Comment (VibhuJawa) [reply python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py]: Thanks. Approved it .

File: python/cugraph-pyg/cugraph_pyg/data/__init__.py [hunk 7]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2019-2022, NVIDIA CORPORATION.
+# Copyright (c) 2019-2023, NVIDIA CORPORATION.
```
Comment (rlratzel) [inline python/cugraph-pyg/cugraph_pyg/data/__init__.py:1]: I didn't catch this in PR 3170: should this year range start at 2022?

**Review by rlratzel — APPROVED** [review]
Noticed one minor item which need not hold up approval.

**Pull request merged**